### PR TITLE
Alerting: Fix removing reducer when inital value is instant

### PR DESCRIPTION
**Pull request by soniaAguilarPeiron**
**What is this feature?**

This PR initialize the alert rule form when creating a new alert rule with the right removal of reduce expression done when using instant query in Prometheus or Loki data source.

We were removing the reduce expression once user updates something in the editor.

**Why do we need this feature?**

This removal should be done from the initial render of the form.

**Who is this feature for?**

All users.

**Which issue(s) does this PR fix?**:

Fixes #

see problem: https://raintank-corp.slack.com/archives/C01LJ5F8NRX/p1732625698749139 

**Special notes for your reviewer:**

After the fix and clicking new alert rule from the list view: 

https://github.com/user-attachments/assets/86bba856-ab29-4449-9adc-d7f0623c052f

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx [hunk 1]
```diff
@@ -154,6 +155,15 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
...
 
+  useEffect(() => {
+    // we only remove or add the reducer(optimize reducer) expression when creating a new alert.
+    // When editing an alert, we assume the user wants to manually adjust expressions and queries for more control and customization.
+
+    if (!editingExistingRule && isOptimizeReducerEnabled) {
+      dispatch(optimizeReduceExpression({ updatedQueries: dataQueries, expressionQueries }));
+    }
+  }, [dataQueries, expressionQueries, editingExistingRule, isOptimizeReducerEnabled]);
+
```
Comment (konrad147) [inline public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:166]: As this should work only for new rules, shouldn't we do that in `getDefaultQueries` instead? 
Doing it in `getDefaultQueries` could lower the complexity of this component

Comment (soniaAguilarPeiron) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:166]: we want to work also while editing the query, not only in the initial render

Comment (konrad147) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:166]: That's true, but we already have code responsible for that, don't we?

Comment (konrad147) [reply public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx:166]: This `useEffect` dispatches `optimizeReduceExpression` on every keystroke. 
I guess we should do that only if the user switches between simple and advanced modes, wdyt?
I'm concerned that invoking that on every key stroke/input change will lead to errors and performance issues

**Review by gillesdemey — APPROVED** [review]

**Review by konrad147 — APPROVED** [review]
LGTM!

**Pull request merged**